backend/tests/integration/pidstore/minters/test_minters_texkey.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+import pytest
+from helpers.utils import create_record, override_config
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+
+from inspirehep.pidstore.errors import PIDAlreadyExistsError
+
+
+def test_minter_mint_new_record_without_texkey(inspire_app):
+    data = {
+        "authors": [{"full_name": "Janeway, K."}],
+        "publication_info": [{"year": 2000}],
+    }
+    with override_config(FEATURE_FLAG_ENABLE_TEXKEY_MINTER=True):
+        record = create_record("lit", data=data)
+
+    expected_pid_value_starts_with = "Janeway:2000"
+    pid = (
+        PersistentIdentifier.query.filter(PersistentIdentifier.object_uuid == record.id)
+        .filter(PersistentIdentifier.pid_type == "texkey")
+        .one()
+    )
+    assert pid.pid_value.startswith(expected_pid_value_starts_with)
+
+
+def test_minter_mint_new_record_with_texkey(inspire_app):
+    data = {
+        "authors": [{"full_name": "Janeway, K."}],
+        "publication_info": [{"year": 2000}],
+        "texkeys": ["Janeway:2000abc"],
+    }
+    with override_config(FEATURE_FLAG_ENABLE_TEXKEY_MINTER=True):
+        record = create_record("lit", data=data)
+
+    expected_pid_value = "Janeway:2000abc"
+    pid = (
+        PersistentIdentifier.query.filter(PersistentIdentifier.object_uuid == record.id)
+        .filter(PersistentIdentifier.pid_type == "texkey")
+        .one()
+    )
+    assert pid.pid_value == expected_pid_value
```

Comment:
I don't think you have a test for the case where the record already has several texkeys in the metadata.

Reply:
:+1: